zebROS_ws/src/controllers_2023/src/elevator_controller.cpp

```diff
@@ -0,0 +1,443 @@
+        //answer to my question: the TalonCommandInterface is passed in if it's not a multiInterfaceController, and just one kind of joint is made!
+        virtual bool init(hardware_interface::RobotHW *hw,
+                          ros::NodeHandle             &root_nh,
+                          ros::NodeHandle             &controller_nh) override;
+        virtual void starting(const ros::Time &time) override;
+        virtual void update(const ros::Time &time, const ros::Duration &period) override;
+        virtual void stopping(const ros::Time &time) override;
+
+        bool cmdService(controllers_2023_msgs::ElevatorSrv::Request &req,
+                        controllers_2023_msgs::ElevatorSrv::Response &res);
+
+
+    private:
+        ros::Time last_time_down_;
+        talon_controllers::TalonControllerInterface elevator_joint_; //interface for the talon joint
+
+        realtime_tools::RealtimeBuffer<ElevatorCommand_2023> position_command_; //this is the buffer for percent output commands to be published
+        ros::ServiceServer elevator_service_; //service for receiving commands
+
+        bool zeroed_;
+        bool last_zeroed_;
+        double last_position_;
+        //double last_setpoint_;
+        hardware_interface::TalonMode last_mode_;
+
+    
+
+
+
+        std::atomic<double> arb_feed_forward_up_high;
+        std::atomic<double> arb_feed_forward_up_low;
+        std::atomic<double> elevator_zeroing_percent_output;
+        std::atomic<double> elevator_zeroing_timeout;
+        std::atomic<double> stage_2_height;
+        std::atomic<double> motion_magic_velocity_fast;
+        std::atomic<double> motion_magic_acceleration_fast;
+        std::atomic<int> motion_s_curve_strength;
+
+        ddynamic_reconfigure::DDynamicReconfigure ddr_;
+
+        
+
+
+
+}; //class
+
+
+ //namespace
+#endif
+
+//END OF HPP CONTENTS
+template<typename T>
+bool readIntoScalar(ros::NodeHandle &n, const std::string &name, std::atomic<T> &scalar){
+    T val;
+	if (n.getParam(name, val)){
+        scalar = val;
+        return true;
+    }
+    return false;
+}
+
+//#include "controllers_2019/elevator_controller.h"
+// ^ not needed since the hpp contents is literally:   controllers_2019/elevator_controller.h
+
+bool ElevatorController_2023::init(hardware_interface::RobotHW *hw,
+                                   ros::NodeHandle             &/*root_nh*/,
+                                   ros::NodeHandle             &controller_nh)
+{
+    //create the interface used to initialize the talon joint
+    hardware_interface::TalonCommandInterface *const talon_command_iface = hw->get<hardware_interface::TalonCommandInterface>();
+
+    //hardware_interface::PositionJointInterface *const pos_joint_iface = hw->get<hardware_interface::PositionJointInterface>();
+
+    
+
+    ddr_.registerVariable<double>
```

Comment:
These should probably be moved after the readIntoScalar code below.  That way the code will have read from the config file before allowing users to dynamically change the values